pyvrp/Population.py

```diff
@@ -87,20 +87,31 @@ def add(self, individual: Individual):
             self.purge()
 
     def remove(self, individual: Individual):
+        """
+        Removes the given individual from the subpopulation. Note that
+        individuals are compared by identity, not by equality.
+
+        Parameters
+        ----------
+        individual
+            The individual to remove.
+
+        Raises
+        ------
+        ValueError
+            When the given individual could not be found in the subpopulation.
+        """
         for _, _, prox in self:  # remove individual from other proximities
             for idx, (other, _) in enumerate(prox):
-                if other == individual:
+                if id(other) == id(individual):
                     del prox[idx]
                     break
 
         for item in self:  # remove individual from subpopulation.
-            if item[0] == individual:
+            if id(item[0]) == id(individual):
```

Comment:
Suggestion: can we change `_Item` from tuple to a dataclass? Then we can be more explicit about what we're accessing from the items: `id(item.individual) == id(individual)`.

Reply:
I do want to keep indexing and unpacking around, so I'll go for a named tuple. That way we get the best of both worlds!